exec/cnex/lib/mmap_posix.c

```diff
@@ -0,0 +1,183 @@
+#include "mmap.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "cell.h"
+#include "exec.h"
+#include "nstring.h"
+#include "number.h"
+#include "object.h"
+#include "stack.h"
+
+
+struct MmapObject {
+    int fd;
+    size_t len;
+    char *view;
+};
+
+
+static struct MmapObject *check_file(TExecutor *exec, Object *pp)
+{
+    if (pp == NULL || pp->ptr == NULL || ((struct MmapObject*)(pp->ptr))->fd < 0) {
+        exec->rtl_raise(exec, "MmapException.InvalidFile", "", BID_ZERO);
+        return NULL;
+    }
+    return pp->ptr;
+}
+
+void object_releaseMMapObject(Object *o)
+{
+    if (o != NULL) {
+        assert(o->refcount > 0);
+        o->refcount--;
+
+        if (o->refcount <= 0) {
+            if (o->ptr != NULL) {
+                struct MmapObject *f = o->ptr;
+                if (f->fd >= 0) {
+                    munmap(f->view, f->len);
+                    close(f->fd);
+                }
+                free(o->ptr);
+            }
+        }
+    }
+}
+
+Cell *object_mmapObjectToString(Object *self)
+{
+    char s[32];
+    snprintf(s, 32, "<mmap %" PRId64 ">", (int64_t)((struct MmapObject*)self->ptr)->fd);
```

Comment:
Use `sizeof(s)` here, don't repeat the number.

Reply:
Fixed all instances of static lengths with `sizeof()`.